File: framework/nexum-core/lib/channel/packets/sync_data_packet.dart
import 'package:nexum_core/models/size.dart';
import 'package:nexum_core/channel/packet.dart';

class SyncDataPacket extends Packet {
  final int fpsLimit;
  final Size screenSize;

  SyncDataPacket({
    required this.fpsLimit,
    required this.screenSize,
  });

  @override
  String toString() => "SyncDataPacket(fpsLimit: $fpsLimit, screenSize: $screenSize)";
}

File: engine/nexum-core/src/main/java/io/nexum/channel/packets/SyncDataPacket.java
package io.nexum.channel.packets;

import io.nexum.models.Size;
import io.nexum.channel.Packet;
import org.jetbrains.annotations.NotNull;

public class SyncDataPacket extends Packet {
    private final int fpsLimit;
    private final @NotNull Size screenSize;

    public SyncDataPacket(int fpsLimit, @NotNull Size screenSize) {
        this.fpsLimit = fpsLimit;
        this.screenSize = screenSize;
    }

    public int getFpsLimit() {
        return fpsLimit;
    }

    public @NotNull Size getScreenSize() {
        return screenSize;
    }
}
